metabase/metabase
# fixes x-ray loading order

Closes https://github.com/metabase/metabase/issues/33085

### Description

Fixes reverse x-ray cards loading order, now they should be loading from top to bottom.

The root cause is [this](https://github.com/metabase/metabase/pull/30630/files#diff-8959e90b56a6d7cccbc5575e5b662017d71fc3fcfb9adc0d4c7079bf8524d327L46) PR, which removes the `dashcards` state which is initialized with `this.getSortedDashcards(props)`. The `dashcard` state prop was not used so it looked like a dead code. But I missed that `getSortedDashcards` actually mutates `props.dashboard.ordered_cards` and since it happens in the constructor, the parent component which is responsible for loading cards data, gets already mutated array of cards in the right order so that is why it worked correctly before.

The E2E test to check x-ray loading order is kinda ugly so any ideas on making it better are very welcomed.

### How to verify

1. Go to http://localhost:3000/browse/1-sample-database
2. Open dev tools, network tab, set throttling to "Slow 3G" for convenience 
3. Open any X-ray and check cards are loaded from top to bottom

### Checklist
